### PR TITLE
Update sfno to latest version

### PR DESCRIPTION
**Pull request by daviddpruitt**
# Modulus Pull Request

## Description
Upate SFNO to latest version

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: modulus/datapipes/climate/sfno/dataloaders/data_loader_multifiles.py [hunk 1]
```diff
@@ -12,133 +12,374 @@
...
 
 import h5py
 import numpy as np
+import torch
 from torch.utils.data import Dataset
 
-# import cv2
-from modulus.utils.sfno.img_utils import reshape_fields
+# can replace this import with zoneinfo from the standard library in python3.9+.
+import pytz
```
Comment (mnabian) [inline modulus/datapipes/climate/sfno/dataloaders/data_loader_multifiles.py:29]: Can we get rid of this new dependency?

**Comment by mnabian** [comment]
@daviddpruitt could you please merge main? Also, please black format the code.

File: modulus/datapipes/climate/sfno/dataloaders/dali_es_helper_2d.py [hunk 2]
```diff
@@ -30,6 +30,9 @@
 import torch
 import zarr
 
+# can replace this import with zoneinfo from the standard library in python3.9+.
+import pytz
```
[the remark this answers is not in the document]
Comment (akshaysubr) [reply modulus/datapipes/climate/sfno/dataloaders/dali_es_helper_2d.py:34]: This is already a dependency in Modulus for some of the TISR calcs

**Comment by mnabian** [comment]
Please also check whether this PR breaks loading the existing checkpoints.

**Review by akshaysubr — APPROVED** [review]

**Review by mnabian — APPROVED** [review]

File: modulus/datapipes/climate/sfno/dataloaders/data_loader_multifiles.py (hunk 1 above)
Comment (daviddpruitt) [reply modulus/datapipes/climate/sfno/dataloaders/data_loader_multifiles.py:28]: Done

**Comment by mnabian** [comment]
Closing as SFNO model will be updated in another PR: https://github.com/NVIDIA/modulus/pull/232

**Pull request closed without merging**